### PR TITLE
Remove JMAP support from Cyrus IMAPd 3.0 release

### PR DESCRIPTION
**Pull request by rsto**
As discussed we want to get rid of the experimental JMAP implementation in 3.0. It's doing installations no favour building on a JMAP codebase that's way outdated in contrast to the current JMAP spec implementation on the master branch.

And it makes our Cassandane and Jenkins tests unnecessarily complex and fiddle.

Please check the commit message if the wording is OK with you?

@nicolan I removed JMAP from all code, but left documentation untouched. E.g. here is a list of affected files that still mention JMAP:

    doc/legacy/changes.html
    docsrc/quickstart.rst
    docsrc/overview/cyrus_roadmap.rst
    docsrc/index.rst
    docsrc/imap/developer/jmap.rst
    docsrc/imap/developer/developer-testing.rst
    docsrc/imap/developer/compiling.rst
    docsrc/imap/reference/faqs/interop-8bit.rst
    docsrc/imap/developer.rst
    docsrc/imap/concepts/features.rst
    docsrc/imap/download/upgrade.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-beta6.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-beta1.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.1.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-beta3.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.2.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-beta4.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.3.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-rc4.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-rc3.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-rc2.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-beta5.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-beta2.rst
    docsrc/imap/download/release-notes/3.0/x/3.0.0-rc1.rst
    docsrc/imap/download/installation/manage-dav.rst
    docsrc/imap/download/installation/http/jmap.rst
    docsrc/imap/download/packagers.rst

The only other places are:

    imap/annotate.c

which is better left untouched and

    perl/annotator/AnnotateInlinedCIDs.pm

which uses JMAP just as an example.

@elliefm Once this PR lands on master, I'll wipe the 3.0-specific Cassandane tests from the test suites.

### CODE REVIEW

**Comment by nicolan** [comment]
@rsto The docs aren't hard to manage: please have a go at removing JMAP docs from the 3.0 branch. I would leave mention of it in the prior 3.0 release notes, but append something saying it's unstable and was removed from 3.0 in 3.0.5

**Comment by rsto** [comment]
@nicolan thanks for the feedback. I just pushed a commit that removes JMAP documentation from docsrc except for developer documentation.

File: docsrc/imap/developer/compiling.rst [hunk 1]
```diff
@@ -183,8 +183,7 @@ Optionally install dependencies for :ref:`building the docs <contribute-docs>`.
 Compile Cyrus
 =============
 
-There are additional :ref:`compile and installation steps<imapinstall-xapian>` if you are using Xapian for searching,
-or if you are :ref:`using jmap <developer-jmap>`.
+There are additional :ref:`compile and installation steps<imapinstall-xapian>` if you are using Xapian for searchin:.
```
Comment (nicolan) [inline docsrc/imap/developer/compiling.rst]: searching.

**Comment by nicolan** [comment]
I'm also hoping that Travis build failure will fix on next commit - looks like a transient fetch problem. Couldn't work out how to force it to just rerun.

File: docsrc/imap/developer/jmap.rst [hunk 2]
```diff
@@ -6,6 +6,14 @@ JMAP support
 
 `JMAP <http://jmap.io/>`_ is an alternate mechanism for synchronising a mail client with a mail server, intended as a replacement for IMAP. It is a transport-agnostic, stateless JSON-based API, aiming to be compatible with the IMAP data model, but with reduced data usage and more efficient synchronisation.
 
+.. warning::
+
+    Cyrus versions 3.0 to 3.0.4 included experimental JMAP support, based on a snapshot of the then current JMAP specification draft. However, since the code-freeze of the Cyrus IMAPd 3.0 release, the JMAP specification at the `IETF JMAP working group <https://datatracker.ietf.org/wg/jmap/about/>` has evolved rapidly. Backporting the latest JMAP specification to Cyrus version 3.0 was unfeasible. Version 3.0.5 removed the outdated JMAP implementation draft from the 3.0 stable branch. Future Cyrus stable releases will implement JMAP once the speficiation is standardized.
```
Comment (elliefm) [inline docsrc/imap/developer/jmap.rst]: Release 3.0.5 also contains the JMAP support

File: docsrc/imap/developer/jmap.rst [hunk 3]
```diff
@@ -6,6 +6,14 @@ JMAP support
...
+
+    Installations currently using JMAP are highly encouraged to upgrade to the latest Cyrus IMAPd development snapshot which implements the latest specification draft.
+
+    The remainder of this section is only relevant for versions 3.0 to 3.0.4.
```
Comment (elliefm) [inline docsrc/imap/developer/jmap.rst]: Ditto re 3.0.5

**Comment by rsto** [comment]
@nicolan @elliefm Could you please approve your request changes? I don't have the right permissions on the repo to approve review changes. This PR should be go to land on the 3.0 branch.

**Comment by elliefm** [comment]
There's still one off-by-one version number in there but I'll just quickly fix it post-merge rather than going through another review roundtrip! :)

**Pull request merged**